plugins/dicom_viewer/plugin_tests/dicom_viewer_test.py

```diff
@@ -22,69 +28,151 @@ class DicomViewerTest(base.TestCase):
+
+        # Check if the 'dicomItem' is well processed
+        dicomItem = Item().load(item['_id'], force=True)
+        self.assertHasKeys(dicomItem, ['dicom'])
+        self.assertHasKeys(dicomItem['dicom'], ['meta'])
+        self.assertHasKeys(dicomItem['dicom'], ['files'])
+
+        # Check if the files list contain the good keys and all the file are well sorted
+        for i in range(0,4):
+            self.assertTrue('_id' in dicomItem['dicom']['files'][i])
+            self.assertTrue('name' in dicomItem['dicom']['files'][i])
+            self.assertEqual(dicomItem['dicom']['files'][i]['name'], '00000%i.dcm' % i)
+            self.assertTrue('SeriesNumber' in dicomItem['dicom']['files'][i]['dicom'])
+            self.assertTrue('InstanceNumber' in dicomItem['dicom']['files'][i]['dicom'])
+            self.assertTrue('SliceLocation' in dicomItem['dicom']['files'][i]['dicom'])
+
+        # Check the common metadata
+        # --> TODO
+        self.assertIsNotNone(dicomItem['dicom']['meta'])
+
+    def testMakeDicomItem(self):
+        admin, user = self.users
 
-        # test filters
-        path = '/item/%s/dicom' % item.get('_id')
-        resp = self.request(path=path, user=admin, params=dict(filters='Rows'))
+        # create a collection, folder, and item
+        collection = Collection().createCollection('collection2', admin, public=True)
+        folder = Folder().createFolder(collection, 'folder2', parentType='collection', public=True)
+        item = Item().createItem('item2', admin, folder)
+
+        # Upload files
+        self._uploadDicomFiles(item, admin)
+
+        # Check the endpoint 'parseDicom' for an admin user
+        dicomItem = Item().load(item['_id'], force=True)
+        dicomItem = self._purgeDicomItem(dicomItem)
+        path = '/item/%s/parseDicom' % dicomItem.get('_id')
+        resp = self.request(path=path, method='POST', user=admin)
         self.assertStatusOk(resp)
-        dicom = resp.json[0]['dicom']
-        self.assertEqual(dicom['Rows'], 80)
-        self.assertEqual(dicom.get('Columns'), None)
-
-        # test non-admin force
-        path = '/item/%s/dicom' % item.get('_id')
-        resp = self.request(path=path, user=user, params=dict(force=True))
+        dicomItem = Item().load(item['_id'], force=True)
+        self.assertHasKeys(dicomItem, ['dicom'])
+        self.assertHasKeys(dicomItem['dicom'], ['meta'])
+        self.assertHasKeys(dicomItem['dicom'], ['files'])
+
+        # Check the endpoint 'parseDicom' for an non admin user
+        dicomItem = Item().load(item['_id'], force=True)
+        dicomItem = self._purgeDicomItem(dicomItem)
+        path = '/item/%s/parseDicom' % dicomItem.get('_id')
+        resp = self.request(path=path, method='POST', user=user)
         self.assertStatus(resp, 403)
+
+    def _uploadDicomFiles(self, item, user):
+        # Upload the files in the reverse order to check if they're well sorted
+        for i in range(0,4):
```

Comment:
To test out-of-order uploads, let's make this `reversed(range(0, 4))`.

Reply:
`range(4, 0, -1)`